alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -580,9 +599,12 @@ class AlertService(
                     }
                 }
                 Alert.State.AUDIT -> {
+                    val index = if (alertIndices.isAlertHistoryEnabled()) {
+                        dataSources.alertsHistoryIndex
```

Comment:
let's create an issue for investigating how to handle migration of data when alert history setting is flipped.

Reply:
created issue.